src/lib/addons/webhook.ts

```diff
@@ -103,16 +103,21 @@ export default class Webhook extends Addon {
             state = 'failed';
             const failedMessage = `Webhook request failed with status code: ${res.status}.`;
             stateDetails.push(failedMessage);
-            if (this.flagResolver.isEnabled('addonUsageMetrics')) {
-                this.eventBus.emit(ADDON_EVENTS_HANDLED, {
-                    result: state,
-                    destination: 'webhook',
-                });
-            }
-
             this.logger.warn(failedMessage);
         }
 
+        if (this.flagResolver.isEnabled('addonUsageMetrics')) {
+            this.eventBus.emit(ADDON_EVENTS_HANDLED, {
+                result: state,
+                destination: 'webhook',
+            });
+        }
+
+        const domain = new URL(url).hostname;
```

Comment:
@daveleek can this throw an exception? I'd recommend we try catch but you can quickly tell me if I'm wrong with a test

Reply:
This should occur after the webhook integration has already run, so if for example the protocol is missing, this integration will fail with an error message before we attempt to log the domain:

> Error querying <url-without-protocol> with method POST status code undefined TypeError: Only HTTP(S) protocols are supported